### PR TITLE
Bugfix/noissue/use try catch to capture canceled exeption

### PR DESCRIPTION
**Pull request by ma12-co**

### CODE REVIEW

File: src/services/messagesService.js [hunk 1]
```diff
@@ -25,21 +25,36 @@ import { generateOcsUrl } from 'nextcloud-router'
...
+	 * cancelToken= the token that gets injected into the axios method and links it
+	 * to the cancel function;
+	 * cancel= function that allows to delete the api call;
+	 */
+	const { token: cancelToken, cancel: cancelFetchMessages } = CANCEL_TOKEN.source()
+	/**
+	 * Fetches messages that belong to a particular conversation
+	 * specified with its token.
+	 *
+	 * @param {string} token The conversation token;
```
Comment (skjnldsv) [inline src/services/messagesService.js]: https://github.com/nextcloud/gallery/blob/refactor/gallery-19/src/utils/CancelableRequest.js
Use this. You'll get much cleaner code on your requests.

Comment (skjnldsv) [reply src/services/messagesService.js]: https://github.com/nextcloud/gallery/blob/6fa4fa6fbafef7dc597927f9b3dd2e581ddade1b/src/components/Folder.vue#L57
https://github.com/nextcloud/gallery/blob/6fa4fa6fbafef7dc597927f9b3dd2e581ddade1b/src/components/Folder.vue#L114-L128

File: src/services/messagesService.js [hunk 2]
```diff
@@ -72,15 +87,15 @@ const cancelableLookForNewMessages = function() {
 		} catch (exception) {
 			if (axios.isCancel(exception)) {
 				console.debug(exception.message)
-				return exception.message
+				throw exception
```
Comment (skjnldsv) [inline src/services/messagesService.js]: You should not catch at all.
No try...catch here

Comment (skjnldsv) [reply src/services/messagesService.js]: https://github.com/nextcloud/gallery/blob/refactor/gallery-19/src/services/FolderInfo.js

File: src/services/messagesService.js (hunk 1 above)
Comment (ma12-co) [reply src/services/messagesService.js]: I am planning use that, but shouldn't it be in another pr?

File: src/components/MessagesList/MessagesList.vue [hunk 3]
```diff
@@ -243,7 +275,7 @@ export default {
 		 */
 		getLastKnownMessageId() {
 			if (this.messagesList[this.messagesList.length - 1]) {
-				return this.messagesList[this.messagesList.length - 1].id
+				return parseInt(this.messagesList[this.messagesList.length - 1].id)
 			}
 			return '0'
```
Comment (nickvergessen) [inline src/components/MessagesList/MessagesList.vue]: above you parse int and here you return a string?
A kitten just died....

**Comment by nickvergessen** [comment]
> ### Conflicting files
> src/components/MessagesList/MessagesList.vue

**Pull request merged**